services/src/main/scala/cromwell/services/cost/GcpCostCatalogService.scala

```diff
@@ -38,28 +39,47 @@ object CostCatalogKey {
   final val expectedSku =
     (".*?N1 Predefined Instance (Core|Ram) .*|" +
       ".*?N2 Custom Instance (Core|Ram) .*|" +
-      ".*?N2D AMD Custom Instance (Core|Ram) .*").r
+      ".*?N2D AMD Custom Instance (Core|Ram) .*|" +
+      "Nvidia Tesla V100 GPU .*|" +
+      "Nvidia Tesla P100 GPU .*|" +
+      "Nvidia Tesla P4 GPU .*|" +
+      "Nvidia Tesla T4 GPU .*").r
+  // TODO: seems like it will probably still match GPU strings with extra stuff in front -
+  // it just won't take any of those preceding characters
+  // What is the point of the .*? ??
 
   def apply(sku: Sku): List[CostCatalogKey] =
     for {
       _ <- expectedSku.findFirstIn(sku.getDescription).toList
-      machineType <- MachineType.fromSku(sku).toList
+      resourceInfo <- ResourceInfo.fromSku(sku).toList
       resourceType <- ResourceType.fromSku(sku).toList
       usageType <- UsageType.fromSku(sku).toList
-      machineCustomization <- MachineCustomization.fromSku(sku).toList
       region <- sku.getServiceRegionsList.asScala.toList
-    } yield CostCatalogKey(machineType, usageType, machineCustomization, resourceType, region)
+      machineCustomization = if (resourceType == Gpu) None else Some(MachineCustomization.fromCpuOrRamSku(sku))
+    } yield CostCatalogKey(resourceInfo, usageType, machineCustomization, resourceType, region)
 
   def apply(instantiatedVmInfo: InstantiatedVmInfo, resourceType: ResourceType): ErrorOr[CostCatalogKey] =
-    MachineType.fromGoogleMachineTypeString(instantiatedVmInfo.machineType).map { mType =>
-      CostCatalogKey(
-        mType,
+    if (resourceType == Gpu)
+      for {
+        gpuInfo <- ErrorOr(instantiatedVmInfo.gpuInfo.get) // TODO: improve error message (default: "None.get")
```

Comment:
Currently, if this method is called with instantiated VM info that does not include GPUs, this line will fail with a very cryptic error message. It should fail — we can't get a GPU cost catalog key for a VM that doesn't use a GPU — but we should improve the error message.